utils/hwloc/hwloc-dump-hwdata-knl.c

```diff
@@ -18,7 +18,7 @@
 
 #define KERNEL_SMBIOS_SYSFS "/sys/firmware/dmi/entries"
 
-#define KNL_SMBIOS_GROUP_STRING "Group: Knights Landing Information"
+#define KNL_SMBIOS_GROUP_STRING "Group: Intel(R) Xeon Phi(TM) Processor Information"
```

Comment:
This string is supposed to be read from SMBIOS tables, I don't think we can change it without breaking things.
But I can't make that change break our regression tests, hwloc-dump-hwdata still finds the KNL SMBIOS group.
@grzegorz-andrejczuk Can you confirm?

Reply:
Yes, these strings are part of DMI entries definition and should not be changed. When the string do not match MCDRAM cache will not be detected. 
Good catch, Thanks.

Reply:
There might be an issue in the current code. When I change the #define'd string, strncmp() doesn't match it anymore, but is_knl_entry() still returns 1: it skips the KNL string that I modified on purpose, and then it tries again on the remaining string, which is now empty, hence strncmp() now matches because we pass len=0.